engine/src/main/java/org/hibernate/search/engine/cfg/impl/EngineConfigurationUtils.java

```diff
@@ -9,29 +9,46 @@
 import java.util.Optional;
 
 import org.hibernate.search.engine.cfg.BackendSettings;
-import org.hibernate.search.engine.cfg.ConfigurationPropertySource;
 import org.hibernate.search.engine.cfg.EngineSettings;
+import org.hibernate.search.engine.cfg.spi.ConfigurationScopeNamespace;
+import org.hibernate.search.engine.cfg.spi.ScopedConfigurationPropertySource;
+import org.hibernate.search.engine.environment.bean.BeanResolver;
 
 public final class EngineConfigurationUtils {
 
 	private EngineConfigurationUtils() {
 	}
 
-	public static ConfigurationPropertySourceExtractor extractorForBackend(Optional<String> backendNameOptional) {
+	public static ConfigurationPropertySourceExtractor extractorForBackend(BeanResolver beanResolver,
+			Optional<String> backendNameOptional) {
 		if ( !backendNameOptional.isPresent() ) {
-			return engineSource -> engineSource.withMask( EngineSettings.Radicals.BACKEND );
+			return engineSource -> engineSource
+					.withScope( beanResolver, ConfigurationScopeNamespace.BACKEND )
+					.withMask( EngineSettings.Radicals.BACKEND );
 		}
 		else {
-			return engineSource -> engineSource.withMask( EngineSettings.Radicals.BACKENDS )
+			return engineSource -> engineSource
+					// First we want to get the defaults for the default backend
+					.withScope( beanResolver, ConfigurationScopeNamespace.BACKEND )
+					// and then on top of that put the backend specific ones (for a particular backend name)
+					.withScope( beanResolver, ConfigurationScopeNamespace.BACKEND, backendNameOptional.get() )
```

Comment:
Are you sure the second call to `withScope` won't just override the first one, i.e. the defaults for the default backend will be ignored?